src/destructuring.rs

```diff
@@ -35,22 +35,16 @@ pub enum LastMatch {
     Ellipsis(Option<Ident>),
 }
 
-/// A destructuring pattern without the `x @` part.
+/// A destructured record pattern
 #[derive(Debug, PartialEq, Clone)]
-pub enum Destruct {
-    /// A record pattern, the only one implemented for now.
-    Record {
-        matches: Vec<Match>,
-        open: bool,
-        rest: Option<Ident>,
-        span: RawSpan,
-    },
-    /// An empty destructuring. In this case, the pattern is a classical `let var = something in
-    /// body` form.
-    Empty,
+pub struct DestructuredRecord {
```

Comment:
Nitpick (naming): `RecordPattern`. Destructuring is the feature but I think pattern is more commonly understood for the thing on the right hand side of the equal sign. That being said, it's a suggestion that you are free to ignore :slightly_smiling_face: